### ISSUE TITLE
Domain support in Identity v3

### ISSUE THREAD

**Issue by stenstad** [fog/fog-openstack-core#44]
With the Identity v2.0 API marked as deprecated in Icehouse (alltho reverted after release), cloud providers are starting to require using the v3 API to authenticate.

This is needed to support domains, to achieve full multi-tenancy in public clouds.

The v2.0 API seems to be scheduled for removal in the Kilo release next year.

Please look into supporting v3 and domains.

Documentation for the v3 API:
- http://api.openstack.org/api-ref-identity-v3.html

Aviator support for v3 API:
- https://github.com/aviator/aviator/blob/master/lib/aviator/openstack/identity/requests/v3/public/create_token.rb

**Comment by clarsonneur** [comment]
Hi Guys. Did you get this patch to support openstack auth v3?

We need it as well.

**Comment by TerryHowe** [comment]
I thought this refactoring effort died out.  Does code in this repo actually get used?

**Comment by clarsonneur** [comment]
May be that's the wrong place. I'm using the gem to get latest fog openstack. 
I noticed v3 not implemented https://github.com/fog/fog/blob/master/lib/fog/openstack/core.rb
and found this request...

**Comment by mikezuniga** [comment]
I'll submit the pull request, however the patch is only for the compute section and getting the v3 tokens inside of core. 

I hope it will be useful to you guys, i was going to refactor identity but i dont need to manage keystone at the moment.

**Comment by plribeiro3000** [comment]
You guys should be looking at `fog`. [openstack](https://github.com/fog/fog/tree/master/lib/fog/openstack) and [h[](https://github.com/fog/fog/tree/master/lib/fog/hp) are not extracted yet. The code here is ready i guess.

**Comment by TerryHowe** [comment]
@mwhagedorn can someone update the README with status maybe?

**Comment by mikezuniga** [comment]
So should i submit the pull request or not?

**Comment by plribeiro3000** [comment]
Yeah. but submit it to `fog` instead. The current code base lies there.

**Comment by plribeiro3000** [comment]
This was a prototype that a i believe is not maintained anymore.

**Comment by TerryHowe** [comment]
https://github.com/fog/fog-openstack-core/pull/46

**Comment by mikezuniga** [comment]
Yup it will be under fog

**Comment by mikezuniga** [comment]
Just leaving this here for reference 

https://github.com/fog/fog/pull/3506
